src/server/routes/rooRoutes.ts

```diff
@@ -704,4 +1027,312 @@ export function registerRooRoutes(
+
+      logger.info(`Updated profile '${name}'`);
+
+      return c.json(
+        {
+          id: updatedId || profileEntry.id,
+          name,
+          message: `Profile '${name}' updated successfully`,
+          activated: activate || false,
+        },
+        200,
+      );
+    } catch (error) {
+      logger.error(`Error updating profile ${c.req.param("name")}:`, error);
+      const message =
+        error instanceof Error ? error.message : "Unknown error occurred";
+      return c.json({ message }, 500);
+    }
+  });
+
+  // DELETE /api/v1/roo/profiles/:name - Delete profile
+  app.openapi(deleteProfileRoute, async (c) => {
+    try {
+      const { name } = c.req.param();
+      const { extensionId } = c.req.query();
+
+      const adapter = controller.getRooAdapter(extensionId);
+      if (!adapter?.isActive) {
+        return c.json({ message: "RooCode extension is not available" }, 500);
+      }
+
+      // Check if profile exists
+      const profileNames = adapter.getProfiles();
+      if (!profileNames.includes(name)) {
+        return c.json({ message: `Profile '${name}' not found` }, 404);
+      }
+
+      // Check if it's the active profile
+      const activeProfileName = adapter.getActiveProfile();
+      if (name === activeProfileName) {
+        return c.json({ message: "Cannot delete the active profile" }, 400);
+      }
+
+      // Delete the profile
+      await adapter.deleteProfile(name);
+
+      logger.info(`Deleted profile '${name}'`);
+
+      return c.json(
+        {
+          message: `Profile '${name}' deleted successfully`,
+        },
+        200,
+      );
+    } catch (error) {
+      logger.error(`Error deleting profile ${c.req.param("name")}:`, error);
+      const message =
+        error instanceof Error ? error.message : "Unknown error occurred";
+      return c.json({ message }, 500);
+    }
+  });
+
+  // GET /api/v1/roo/profiles/active - Get active profile
+  app.openapi(getActiveProfileRoute, async (c) => {
+    try {
+      const { extensionId } = c.req.query();
+
+      const adapter = controller.getRooAdapter(extensionId);
+      if (!adapter?.isActive) {
+        return c.json({ message: "RooCode extension is not available" }, 500);
+      }
+
+      const activeProfileName = adapter.getActiveProfile();
+
+      if (!activeProfileName) {
+        return c.json(
+          {
+            name: undefined,
+            profile: undefined,
+          },
+          200,
+        );
+      }
+
+      const profileEntry = adapter.getProfileEntry(activeProfileName);
+
+      return c.json(
+        {
+          name: activeProfileName,
+          profile: profileEntry,
+        },
+        200,
+      );
+    } catch (error) {
+      logger.error("Error getting active profile:", error);
+      const message =
+        error instanceof Error ? error.message : "Unknown error occurred";
+      return c.json({ message }, 500);
+    }
+  });
+
```

Comment:
This API could not work since it may try to get the profile called "active". I think GET /roo/profiles is enough for the functionality.

```suggestion
```